File: algorithms/lab3/bogosort/bogosort.py
from random import *
from time import *


def correct_order(data):
    count = 0
    while count + 1 < len(data):
        if data[count] > data[count + 1]:
            return False
        count += 1
    return True


def bogosort(data):
    while not correct_order(data):
        shuffle(data)   # randomize data
    return data


if __name__ == "__main__":

    data = [randint(0, 100) for i in range(10)]
    start = time()
    print("Before: ", data)
    data = bogosort(data)
    print("After: ", data)
    print("%.2f seconds" % (time() - start))

File: algorithms/lab3/bogosort/bogosort.js
function shuffle(data){
    var count = data.length, temp, index;

    while(count > 0) {
        index = ~~(Math.random() * count);
        count--;

        temp = data[count];
        data[count] = data[index];
        data[index] = temp;
    }

    return data;
}

function bogosort(data) {
  while(!correct_order(data)) {
    data = shuffle(data);
  }

  return data;
}

function correct_order(data) {
  let sorted = true;

  for(let i = 0; i < data.length -1; i++) {
    if(data[i + 1] < data[i]) {
      sorted = false;
    }
  }

  return sorted;
}

const data = [7, 4, 3, 8, 2, 1, 5, 9];
console.log('Before:', data);
console.log('After:', bogosort(shuffle(data)));
